### PR TITLE
Feature/add post page

### PR DESCRIPTION
**Pull request by barrydilan**
## What type of PR is this? (select all that apply)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 🚧 Breaking Change
- [ ] 🧑‍💻 Code Refactor
- [ ] 📝 Documentation Update

## Related Tickets & Documents

- Related Issue #
- Closes #

## Screenshots, Recordings
Снимок экрана 2023-10-17 в 21 54 43

_Please replace this line with any relevant images for UI changes._

## Added/updated tests?

- [ ] 👌 Yes
- [ ] 🙅‍♂️ No, because they aren't needed
- [ ] 🙋‍♂️ No, because I need help

### CODE REVIEW

File: pages/posts/[slug].tsx [hunk 1]
```diff
@@ -31,13 +64,41 @@ const PostPage = ({ postMetadata, content }: postData) => {
...
+              <p>img: {imageUrl}</p>
             </header>
-            <hr />
-            <MdToHtml mdSource={content} />
+            <hr /> */}
+            <div>
+              <div className="min-h-[280px] w-full relative pl-6 pt-6 pr-24 md:rounded-3xl rounded-none sm:pr-96 sm:pt-12 flex flex-col gap-y-6 md:gap-y-8">
+                <div className="w-full h-full absolute inset-x-0 top-0 bg-black/50 z-10 md:rounded-3xl rounded-none"></div>
+                  <img src="/assets/icons/arrowBack.svg" className="relative z-20 w-4 h-5" alt="" />
+                  <Image src={image} alt={slug} fill objectFit='cover' className='md:rounded-3xl rounded-none'/>
```
Comment (Tedzury) [inline pages/posts/[slug].tsx]: At mobile screens main image should have rounded bottom corners.

File: styles/globals.css [hunk 2]
```diff
@@ -134,3 +134,31 @@ article {
...
+
+#directions img {
+  @apply block h-auto w-full mt-6 mx-auto sm:w-1/2
+}
+
+ul li {
+  @apply flex gap-x-2 items-center;
+}
+
+.active {
```
Comment (Tedzury) [inline styles/globals.css:150]: Text color should be orange, like border bottom.

File: _source/_posts/recipeTest.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+description: "Для тех, кто с утра перед работой любит хорошо покушать, посоветую быстрый способ приготовления домашних сырников"
+slug: recipe
+---
+
+<div class="recipe-buttons">
+  <button class="inline-block text-red-800 active" id="ingridients_btn">Ingridients</button>
+  <button class="inline-block" id="directions_btn">Directions</button>
+</div>
+
+<div id="ingridients" className="">
```
Comment (Tedzury) [inline _source/_posts/recipeTest.md]: Probably, ingridients must be a flex container, and "checkmark" class must have a 50% width at desktop, to place ingridients in two columns. Also, I think we should rename "ingridients" and "directions" into "ингридиенты" and "рецепт"/"готовка".

**Review by Tedzury — CHANGES_REQUESTED** [review]
Nice job! But, I think it needs few improvements.

File: _source/_posts/recipeTest.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+* <span class="checkmark"></span> Яйца
+* <span class="checkmark"></span> Сахар
+* <span class="checkmark"></span> Мука
+</div>
+
+<div id="directions" className="hidden">
+1. <div>Смешать все ингредиенты в любом порядке.<img src="/assets/img/recipeStep.jpeg"/></div>
+2. <div>Столовой ложкой выкладываю порционно на сковороду.<img src="/assets/img/recipeStep.jpeg"/></div>
+3. <div> Обжариваю с каждой стороны по 8-10 минут.<img src="/assets/img/recipeStep.jpeg"/></div>
+</div>
```
Comment (Quiddlee) [inline _source/_posts/recipeTest.md]: Number should be with accent color.

You can place numbers in span and give them class "text-accent-600"

```tsx
<div id="directions" className="hidden">
<div><span className="text-accent-600">1.</span>  Смешать все ингредиенты в любом порядке.<img src="/assets/img/recipeStep.jpeg"/></div>
<div><span className="text-accent-600">2.</span>  Столовой ложкой выкладываю порционно на сковороду.<img src="/assets/img/recipeStep.jpeg"/></div>
<div><span className="text-accent-600">3.</span>   Обжариваю с каждой стороны по 8-10 минут.<img src="/assets/img/recipeStep.jpeg"/></div>
</div>

```

File: _source/_posts/recipeTest.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+* <span class="checkmark"></span> Творог
+* <span class="checkmark"></span> Яйца
+* <span class="checkmark"></span> Сахар
+* <span class="checkmark"></span> Мука
+</div>
+
+<div id="directions" className="hidden">
+1. <div>Смешать все ингредиенты в любом порядке.<img src="/assets/img/recipeStep.jpeg"/></div>
+2. <div>Столовой ложкой выкладываю порционно на сковороду.<img src="/assets/img/recipeStep.jpeg"/></div>
+3. <div> Обжариваю с каждой стороны по 8-10 минут.<img src="/assets/img/recipeStep.jpeg"/></div>
```
Comment (Quiddlee) [inline _source/_posts/recipeTest.md]: images should be rounded. Remove border and increase roundness. Also image should have shadow

File: _source/_posts/recipeTest.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+weight: 23
+image: "/assets/img/syrniki.jpg"
+description: "Для тех, кто с утра перед работой любит хорошо покушать, посоветую быстрый способ приготовления домашних сырников"
+slug: recipe
+---
+
+<div class="recipe-buttons">
+  <button class="inline-block text-red-800 active" id="ingridients_btn">Ingridients</button>
+  <button class="inline-block" id="directions_btn">Directions</button>
+</div>
```
Comment (Quiddlee) [inline _source/_posts/recipeTest.md]: When tab is active text should be accent too

File: components/layouts/postPageLayout.tsx [hunk 7]
```diff
@@ -27,18 +29,18 @@ const PostPageLayout = ({ postMetadata, children }: propsType) => {
...
+      </header> */}
+      {/* <Navbar /> */}
+      {/* <Breadcrumb breadcrumbs={breadcrumbs} /> */}
       <main>
         <div className="flex-grow">{children}</div>
       </main>
-      <div className="container mx-auto flex flex-row items-center justify-between rounded-md text-center text-black shadow-md">
+      {/* <div className="container mx-auto flex flex-row items-center justify-between rounded-md text-center text-black shadow-md">
         <div>
           <DateFormatter dateString={date} />
```
Comment (Quiddlee) [inline components/layouts/postPageLayout.tsx]: Comments can be removed

File: components/layouts/postsIndexLayout.tsx [hunk 8]
```diff
@@ -13,7 +13,7 @@ const PostsIndexLayout = ({ children }: propsType) => {
           <p className="mt-2 text-lg">{webSiteTitle}</p>
         </div>
       </header>
-      <Navbar />
+      {/* <Navbar /> */}
```
Comment (Quiddlee) [inline components/layouts/postsIndexLayout.tsx]: Commented code

File: pages/posts/[slug].tsx [hunk 9]
```diff
@@ -1,25 +1,58 @@
...
+      e.target.classList.add("active");
+      directionsBtn.current?.classList.remove("active");
+      ingridientsDiv.current?.classList.add("block");
+      ingridientsDiv.current?.classList.remove("hidden");
+      directionsDiv.current?.classList.remove("block");
+      directionsDiv.current?.classList.add("hidden");
+    }
+
+    if (e.target.id === "directions_btn") {
+      e.target.classList.add("active");
```
Comment (Quiddlee) [inline pages/posts/[slug].tsx]: Too many eslint errors

File: styles/globals.css [hunk 10]
```diff
@@ -54,9 +54,9 @@ a:hover {
 }
 
 article {
-  padding: 1rem;
-  border: 1px solid #ddd;
-  border-radius: 4px;
+  /* padding: 1rem; */
+  /* border: 1px solid #ddd; */
+  /* border-radius: 4px; */
 }
```
Comment (Quiddlee) [inline styles/globals.css]: Commented code

File: tailwind.config.js [hunk 11]
```diff
@@ -63,7 +63,11 @@ module.exports = {
         'black-800': '#454545',
         'black-900': '#3d3d3d',
         'black-950': '#000000',
+        'active': '#dd4c12',
       },
+      backgroundImage: {
+        checkmark: 'url("/assets/icons/checkIcon.svg")',
+      }
     },
```
Comment (Quiddlee) [inline tailwind.config.js]: We already have active color: "accent-600"

**Review by Quiddlee — CHANGES_REQUESTED** [review]

File: tailwind.config.js (hunk 11 above)
Comment (barrydilan) [reply tailwind.config.js]: updated

File: styles/globals.css (hunk 10 above)
Comment (barrydilan) [reply styles/globals.css]: removed

File: pages/posts/[slug].tsx (hunk 9 above)
Comment (barrydilan) [reply pages/posts/[slug].tsx]: fixed

File: components/layouts/postPageLayout.tsx (hunk 7 above)
Comment (barrydilan) [reply components/layouts/postPageLayout.tsx]: removed

File: components/layouts/postsIndexLayout.tsx (hunk 8 above)
Comment (barrydilan) [reply components/layouts/postsIndexLayout.tsx]: removed

File: _source/_posts/recipeTest.md (hunk 6 above)
Comment (barrydilan) [reply _source/_posts/recipeTest.md]: done

File: _source/_posts/recipeTest.md (hunk 5 above)
Comment (barrydilan) [reply _source/_posts/recipeTest.md]: done

File: _source/_posts/recipeTest.md (hunk 4 above)
Comment (barrydilan) [reply _source/_posts/recipeTest.md]: done

File: _source/_posts/recipeTest.md (hunk 3 above)
Comment (barrydilan) [reply _source/_posts/recipeTest.md]: its now display:grid starting from small screens

File: pages/posts/[slug].tsx (hunk 1 above)
Comment (barrydilan) [reply pages/posts/[slug].tsx]: It looks good in figma but its not pretty when combined with full layout - looks out of place, thats why i have made the image rounded only on tablet and up

File: components/layouts/postPageLayout.tsx [hunk 12]
```diff
@@ -15,7 +15,9 @@ type propsType = {
 };
 
 const PostPageLayout = ({ postMetadata, children }: propsType) => {
-  const { title, subtitle, category, slug, date, author } = postMetadata;
+  const { title, subtitle, category, slug, date, author, image } = postMetadata;
+  console.log(image);
+  console.log(postMetadata);
   const breadcrumbs: BreadcrumbObj[] = [
```
Comment (Quiddlee) [inline components/layouts/postPageLayout.tsx]: console.log

**Review by Quiddlee — COMMENTED** [review]
Back btn is not working. Also problems with recipe image borders, i think borders should be complitely removed

**Review by Quiddlee — COMMENTED** [review]
Back btn is not working. Also problems with recipe image borders

**Comment by barrydilan** [comment]
> Back btn is not working. Also problems with recipe image borders

Back btn is fixed. Cannot replicate border behavior, can you check again?

**Review by Quiddlee — APPROVED** [review]

**Review by Tedzury — APPROVED** [review]

**Pull request merged**